File: frontend/src/types/nutritionLimit.ts
export interface NutritionLimit {
  id: number
  date: string
  calories_burned: number | null
  body_weight: number | null
  activity_level: string | null
  calories: number | null
  proteins: number | null
  carbohydrates: number | null
  carbohydrates_of_sugars: number | null
  fats: number | null
  fats_saturated: number | null
  salt: number | null
  fibers: number | null
}

export interface NutrientLimitsPreview {
  calories: number
  proteins: number
  carbohydrates: number
  carbohydrates_of_sugars: number
  fats: number
  fats_saturated: number
  salt: number
  fibers: number
}

export interface NutritionLimitCreate {
  date: string
  calories_burned?: number | null
  body_weight?: number | null
  activity_level?: string | null
  calories?: number | null
  proteins?: number | null
  carbohydrates?: number | null
  carbohydrates_of_sugars?: number | null
  fats?: number | null
  fats_saturated?: number | null
  salt?: number | null
  fibers?: number | null
}

export interface NutritionLimitUpdate {
  calories_burned?: number | null
  body_weight?: number | null
  activity_level?: string | null
  calories?: number | null
  proteins?: number | null
  carbohydrates?: number | null
  carbohydrates_of_sugars?: number | null
  fats?: number | null
  fats_saturated?: number | null
  salt?: number | null
  fibers?: number | null
}

export interface PreviewRequest {
  calories_burned: number
  body_weight: number
  activity_level: string
}

File: backend/app/schemas/nutrition_limit.py
from datetime import date as date_type

from pydantic import BaseModel, Field


class NutrientLimitsPreviewRequest(BaseModel):
    calories_burned: float = Field(gt=0)
    body_weight: float = Field(gt=0, le=500)
    activity_level: str


class NutrientLimitsPreview(BaseModel):
    calories: float
    proteins: float
    carbohydrates: float
    carbohydrates_of_sugars: float
    fats: float
    fats_saturated: float
    salt: float
    fibers: float


class NutritionLimitCreate(BaseModel):
    date: date_type
    calories_burned: float | None = None
    body_weight: float | None = None
    activity_level: str | None = None
    calories: float | None = None
    proteins: float | None = None
    carbohydrates: float | None = None
    carbohydrates_of_sugars: float | None = None
    fats: float | None = None
    fats_saturated: float | None = None
    salt: float | None = None
    fibers: float | None = None


class NutritionLimitUpdate(BaseModel):
    calories_burned: float | None = None
    body_weight: float | None = None
    activity_level: str | None = None
    calories: float | None = None
    proteins: float | None = None
    carbohydrates: float | None = None
    carbohydrates_of_sugars: float | None = None
    fats: float | None = None
    fats_saturated: float | None = None
    salt: float | None = None
    fibers: float | None = None


class NutritionLimitRead(BaseModel):
    id: int
    date: date_type
    calories_burned: float | None
    body_weight: float | None
    activity_level: str | None
    calories: float | None
    proteins: float | None
    carbohydrates: float | None
    carbohydrates_of_sugars: float | None
    fats: float | None
    fats_saturated: float | None
    salt: float | None
    fibers: float | None

    model_config = {"from_attributes": True}


class NutritionLimitListResponse(BaseModel):
    items: list[NutritionLimitRead]
    total: int
